### PR TITLE
feat: add cfn-lint

### PR DESCRIPTION
**Pull request by vishnu-deepsource**

### CODE REVIEW

File: analyzers/cfn-lint/.deepsource/analyzer/example_config.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+version = 1
+
+[[analyzers]]
+name = "cfn-lint"
```
Comment (srijan-deepsource) [inline analyzers/cfn-lint/.deepsource/analyzer/example_config.toml:4]:
```suggestion
name = "cfn-lint"
type = "community"
```

File: analyzers/cfn-lint/.deepsource/analyzer/issues/CFLIN-E0002.toml
Comment (srijan-deepsource) [inline analyzers/cfn-lint/.deepsource/analyzer/issues/CFLIN-E0002.toml:1]: issues folder should be outside. `.deepsource/issues`

File: analyzers/cfn-lint/utils/issue_gen.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                            all_classes_data.append(class_data)
+                    except Exception as e:
+                        print(f"Error parsing file {file}: {e}")
+
+    return all_classes_data
+
+
+if __name__ == "__main__":
+    base_directory = "/Users/vishnu/Code/deepsource/"
+    rules_directory = "./cfn-lint/src/cfnlint/rules"
```
Comment (srijan-deepsource) [inline analyzers/cfn-lint/utils/issue_gen.py]: Remove hardcoded values? We can take this in as input.

File: analyzers/cfn-lint/.deepsource/analyzer/issues/CFLIN-W8001.toml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+title = "Check if Conditions are Used"
```
Comment (srijan-deepsource) [inline analyzers/cfn-lint/.deepsource/analyzer/issues/CFLIN-W8001.toml]:
```suggestion
title = "Unused conditions detected"
```

File: analyzers/cfn-lint/.deepsource/analyzer/issues/CFLIN-W8003.toml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+title = "Fn::Equals will always return true or false"
```
Comment (srijan-deepsource) [inline analyzers/cfn-lint/.deepsource/analyzer/issues/CFLIN-W8003.toml]:
```suggestion
title = "`Fn::Equals` will always return true or false"
```

**Review by srijan-deepsource — CHANGES_REQUESTED** [review]
I see that a good number of titles are in title-case. We use sentence-case for titles. Can you please take care of them?

Also, add a test for cfn lint's sarif file to make sure our sarif-parser works properly with this.
Ref: https://github.com/DeepSourceCorp/community-analyzers/tree/master/tests

File: analyzers/cfn-lint/CI/github-actions.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+          cfn-lint -t ./**/*.yaml -f sarif > cfn-lint.sarif
+        continue-on-error: true
+
+      - name: Upload SARIF report files to DeepSource
+        run: |
+          # Install the CLI
+          curl https://deepsource.io/cli | sh
+
+          # Send the report to DeepSource
+          ./bin/deepsource report --analyzer cfn-lint --value-file ./cfn-lint.sarif
```
Comment (eshaan-deepsource) [inline analyzers/cfn-lint/CI/github-actions.yaml]:
```suggestion
          ./bin/deepsource report --analyzer cfn-lint --analyzer-type community --value-file ./cfn-lint.sarif
```

File: analyzers/cfn-lint/.deepsource/issues/CFLIN-E0002.toml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+title = "error processing rule on the template"
```
Comment (eshaan-deepsource) [inline analyzers/cfn-lint/.deepsource/issues/CFLIN-E0002.toml]: Let's make all the titles capitilized.

**Comment by eshaan-deepsource** [comment]
Can you add an entry for cfn-lint in the [README.md](https://github.com/DeepSourceCorp/community-analyzers?tab=readme-ov-file#supported-analyzers) as well?

**Review by eshaan-deepsource — APPROVED** [review]

**Pull request merged**